File: day1/task1.hs
import Data.List (sort, zip)

readNums :: String -> (Int, Int)
readNums s =
  let (n, s') = head (reads s :: [(Int, String)])
      (n', _) = head (reads s' :: [(Int, String)])
   in (n, n')

readLines :: [String] -> ([Int], [Int])
readLines [] = ([], [])
readLines (x : xs) =
  let (n, n') = readNums x
      (ns, ns') = readLines xs
   in (n : ns, n' : ns')

distance :: [Int] -> [Int] -> Int
distance xs xs' = foldl (\acc (x, y) -> acc + abs (x - y)) 0 (zip (sort xs) (sort xs'))

main = do
  contents <- readFile "input.txt"
  let (l, l') = readLines $ lines contents
  print $ distance l l'

File: day1/task1.py
with open("input.txt", "r") as f:
    nums = [l.split("   ") for l in f.readlines()]

l1 = sorted([int(n) for n, _ in nums])
l2 = sorted([int(n) for _, n in nums])
res = sum(abs(n1 - n2) for n1, n2 in zip(l1, l2))
print(res)
